### PR TITLE
Release 08/14/2017

### PR DESCRIPTION
**Pull request by adelevie**

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -25,52 +24,31 @@ func main() {
...
 	o.HandleFunc("/{service}/callback", handlers.AuthCallbackHandler)
 
-	// Validation
-	v := r.PathPrefix("/validate").Subrouter()
-	v.HandleFunc("/ssn/{ssn}", handlers.ValidateSSN)
-	v.HandleFunc("/email", handlers.ValidateEmail)
+	// Account specific actions
+	a := r.PathPrefix("/me").Subrouter().Inject(handlers.JwtTokenValidatorHandler)
+	a.HandleFunc("/validate", handlers.Validate).Methods("POST")
+	a.HandleFunc("/save", handlers.Save).Methods("POST")
```
Comment (adelevie) [inline api/main.go]: should this be PUT? does it matter?

File: api/model/form/location.go [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+	case LayoutBirthPlaceWithoutCounty:
+		if domestic {
+			stack = validateFields(entity, "city", "state")
+		} else {
+			stack = validateFields(entity, "city", "county")
+		}
+	case LayoutCountry:
+		stack = validateFields(entity, "country")
+	case LayoutUSCityStateInternationalCity:
+		if domestic {
```
Comment (adelevie) [inline api/model/form/location.go]: this if/else pattern and the `validateFields` lines get repeated a lot. maybe a good candidate to refactor.

File: api/model/form/location.go [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+
+	return !stack.HasErrors(), stack
+}
+
+func validateFields(entity *Location, props ...string) model.ErrorStack {
+	var stack model.ErrorStack
+
+	for _, prop := range props {
+		switch strings.ToLower(prop) {
+		case "street":
```
Comment (adelevie) [inline api/model/form/location.go]: these might be refactorable.

File: api/model/form/military.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+	Explanation        Payload
+}
+
+// Unmarshal bytes in to the entity properties.
+func (entity *MilitarySelective) Unmarshal(raw []byte) error {
+	return json.Unmarshal(raw, entity)
+}
+
+// Valid checks the value(s) against an battery of tests.
+func (entity *MilitarySelective) Valid() (bool, error) {
```
Comment (adelevie) [inline api/model/form/military.go]: the logic here gets a little gnarly ... another refactor candidate?

File: api/main.go (hunk 1 above)
Comment (bmallred) [inline api/main.go]: Good question. We should actually listen on both (for at least `/save`). This should be compliant  to standards of HTTP 1.1 and 2.0.

Funny thing about 2.0 is they are talking about doing away with PUT and DELETE verbs.

File: api/model/form/military.go (hunk 4 above)
Comment (bmallred) [inline api/model/form/military.go]: Could do some early exiting in place of the `if-else` branches.

File: api/model/form/location.go (hunk 2 above)
Comment (bmallred) [inline api/model/form/location.go]: We'll take a look at it. Pretty much copied the logic/patterns applied on the JavaScript side.

This one may be ugly just by the nature of how many different combinations there are, too.

File: src/schema/form/textarea.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export default const textarea = {
```
Comment (adelevie) [inline src/schema/form/textarea.json]: should the file extension be JS and not JSON?

Comment (bmallred) [inline src/schema/form/textarea.json]: Absolutely! This one is being ingested yet but that is a great find

**Pull request merged**